app/src/main/java/com/greenfox/kalendaryo/MainActivity.java

```diff
@@ -1,13 +1,167 @@
+    private TextView myText;
+    private static final int REQ_CODE = 900;
+    private SharedPreferences sharedPref;
+    private SharedPreferences.Editor editor;
+    private GoogleSignInAccount account;
 
-public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        profileSection = findViewById(R.id.prof_section);
+        signOut = findViewById(R.id.bn_logout);
+        signIn = findViewById(R.id.bn_login);
+        loginName = findViewById(R.id.name);
+        loginEmail = findViewById(R.id.email);
+        signIn.setOnClickListener(this);
+        signOut.setOnClickListener(this);
+        profileSection.setVisibility(View.GONE);
+        myText = findViewById(R.id.myText);
+        TextView token = findViewById(R.id.tokenText);
+        findViewById(R.id.button2).setOnClickListener(this);
+
+        sharedPref = getSharedPreferences("userInfo", Context.MODE_PRIVATE);
+        String username = sharedPref.getString("username","");
+        String useremail = sharedPref.getString("useremail","");
+        Log.d("sharedpref", username);
+                if(username.equals(null)){
+            GoogleSignInOptions signInOptions = new GoogleSignInOptions
+                    .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                    .requestEmail()
+                    .requestIdToken("141350348735-cibla76rafmvq6c6enon40kc6eg3r9su.apps.googleusercontent.com")
+                    .requestServerAuthCode("141350348735-cibla76rafmvq6c6enon40kc6eg3r9su.apps.googleusercontent.com")
+                    .build();
+            googleApiClient = new GoogleApiClient
+                    .Builder(this)
+                    .enableAutoManage(this,this)
+                    .addApi(Auth.GOOGLE_SIGN_IN_API, signInOptions)
+                    .build();
+        } else {
+            settingDisplayNameAndEamil(username, useremail);
+            updateUI(true);
+        }
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.bn_login:
+                signIn();
+                break;
+            case R.id.bn_logout:
+                signOut();
+                break;
+            case R.id.button2:
+                displayData();
+                break;
+        }
+    }
+
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+      
+    }
+
+    public void signIn() {
+        Intent intent = Auth.GoogleSignInApi.getSignInIntent(googleApiClient);
+        startActivityForResult(intent, REQ_CODE);
+    }
+
+    public void signOut() {
+        Auth.GoogleSignInApi.signOut(googleApiClient).setResultCallback(new ResultCallback<Status>() {
+            @Override
+            public void onResult(@NonNull Status status) {
+                updateUI(false);
```

Comment:
This should delete the shared preferences simply, without signing out from the program.